app/scripts/directives/notifications/notificationDrawerWrapper.js

```diff
@@ -0,0 +1,288 @@
+        var notificationListener;
+
+        var eventsWatcher;
+        var eventsFromNotifications = [];
+
+        // TODO:
+        // include both Notifications & Events,
+        // rather than destroying the map each time maintain it & add new items
+        var notificationGroupsMap = {};
+        var notificationGroups = [];
+
+        var projects = {};
+
+        var listProjects = function() {
+          return DataService.list("projects", {}, function(items) {
+            projects = items.by("metadata.name");
+          });
+        };
+
+        var ensureProjectGroupExists = function(groups, projectName) {
+          if(projectName && !groups[projectName]) {
+            groups[projectName] = {
+              heading: $filter('displayName')(projects[projectName]) || projectName,
+              project: projects[projectName],
+              notifications: []
+            };
+          }
+        };
+
+        var deregisterEventsWatch = function() {
+          if(eventsWatcher) {
+            DataService.unwatch(eventsWatcher);
+          }
+        };
+
+        var watchEvents = function(projectName, cb) {
+          if(projectName && ENABLE_EVENT_WATCH) {
+            eventsWatcher = DataService.watch('events', {namespace: projectName}, _.debounce(cb, 400), { skipDigest: true });
+          }
+        };
+
+        // NotificationService notifications are minimal, they do no necessarily contain projectName info.
+        // ATM tacking this on via watching the current project.
+        var watchNotifications = function(projectName, cb) {
+          if(!projectName) {
+            return;
+          }
+          notificationListener = $rootScope.$on('NotificationsService.onNotificationAdded', cb);
+        };
+
+        var deregisterNotificationListener = function() {
+          notificationListener && notificationListener();
+        };
+
+        var unread = function(notifications) {
+          return _.filter(notifications, 'unread');
+        };
+
+        // returns a count for each type of notification, example:
+        // {Normal: 1, Warning: 5}
+        var countUnreadNotificationsForGroup = function(group) {
+          return _.countBy(unread(group.notifications), function(notification) {
+            return notification.status;
+          });
+        };
+
+        // currently we only show 1 at a time anyway
+        var countUnreadNotificationsForAllGroups = function() {
+          _.each(drawer.notificationGroups, function(group) {
+            group.counts = countUnreadNotificationsForGroup(group);
+          });
+        };
+
+        var countAllUnread = function(group) {
+          return _.reduce(group.counts, function(result, count) {
+            return result + count;
+          }, 0);
+        };
+
+        var sortNotifications = function(notifications) {
+          return _.sortBy(notifications, function(notification) {
+            return -(new Date(notification.lastTimestamp));
+          });
```

Comment:
This is going to be called a lot. Since we can sort these as strings given the date format -- `"2017-08-07T17:04:20Z"` -- and we already do it in other places, I'd just do a string comparison and avoid parsing as a date. Changing the API date format would be a breaking change, so we should be safe.

I think this can just be... (untested)

```js
return _.orderBy(notifications, ['lastTimestamp'], ['desc']);
```

But it probably makes sense to have a secondary sort on `firstTimestamp` as well.

```js
return _.orderBy(notifications, ['lastTimestamp', 'firstTimestamp'], ['desc', 'desc']);
```